bin/generate_inputs.py

```diff
@@ -0,0 +1,221 @@
+    allowed_library_source = ['METAGENOMIC']
+    run_fields = 'library_source,library_strategy,fastq_ftp'
+    if input_scientific_name:
+        input_scientific_name = [s.strip() for s in input_scientific_name.split(',')]
+        run_fields += ',scientific_name'
+    if input_env_biome:
+        run_fields += ',environment_biome'
+        input_input_env_biomes = [s.strip() for s in input_env_biome.split(',')]
+    runs = handler.get_study_runs(raw_reads_study, fields=run_fields)
+    print(f'Received {len(runs)} runs from {raw_reads_study}')
+    list_runs = []
+    run_paths = {}
+    for run in runs:
+        if run['library_strategy'] == 'AMPLICON':
+            continue
+        if input_scientific_name:
+            if run['scientific_name'] not in input_scientific_name:
+                continue
+        if input_env_biome:
+            if run['environment_biome'] not in input_input_env_biomes:
+                continue
+        if keep_metat:
+            allowed_library_source.append('METATRANSCRIPTOMIC')
+        if run['library_source'] not in allowed_library_source:
+            continue
+        list_runs.append(run['run_accession'])
+        run_paths[run['run_accession']] = run['fastq_ftp']
+    print(f'Received {len(list_runs)} runs after filtering')
+    assembly_run = {}
+    assemblies = handler.get_study_assemblies(assembly_study, fields='submitted_ftp,generated_ftp')
+    print(f'Received {len(assemblies)} assemblies from {assembly_study}')
+    with open(os.path.join(outdir, 'runs_assemblies.tsv'), 'w') as file_out:
+        for assembly in assemblies:
+            retrieved_run = assembly['submitted_ftp'].split('/')[-1].split('.')[0].split('_')[0]
+            if not retrieved_run.startswith(('ERR', 'DRR', 'SRR')):
+                print('Invalid run name {} for assembly {}'.format(run_acc, erz_acc))
+                sys.exit(1)
+            if retrieved_run not in list_runs:
+                continue
+            file_out.write('\t'.join([retrieved_run, assembly['analysis_accession']]) + '\n')
+            assembly_run[assembly['analysis_accession']] = retrieved_run
+
+            write_samplesheet_line(
+                assembly=assembly['analysis_accession'],
+                assembly_path=assembly['generated_ftp'],
+                run=retrieved_run,
+                run_path=run_paths[retrieved_run].split(';'),
+                samplesheet=samplesheet_path
+            )
+    return assembly_run
+
+
+def main(args):
+    if not os.path.exists(args.outdir):
+        os.mkdir(args.outdir)
+    # add header to input samplesheet
+    samplesheet_path = os.path.join(args.outdir, args.output_samplesheet)
+    with open(samplesheet_path, 'w') as file_out:
+        file_out.write(','.join(['id', 'fastq_1', 'fastq_2', 'assembly_accession', 'assembly_fasta', 'assembler']) + '\n')
+
+    # filter runs
+    assembly_run_dict = generate_lists(
```

Comment:
Nitpick comment. The method `generate_lists` should be either renamed or split, it's not only generating the lists but also writing those to the samplesheet.